### PR TITLE
e2e: Write node Docker container logs to a file

### PR DESCRIPTION
**Pull request by anhductn2001**
Closes: #4045 

## What is the purpose of the change

> Write node Docker container logs to a file, persist as artifacts in CI

## Brief Changelog

  - *Write the docker container logs to a temporary file on the host. Then, persists these log files as artifacts in CI.*
  - *Use `make test-e2e-debug` to skip cleanup container*

## Testing and Verifying
  - *I have tested in fork version [here](https://github.com/anhductn2001/osmosis/actions/runs/4022809583/jobs/6912940242)

### CODE REVIEW

**Comment by p0mvn** [comment]
Thanks for the PR @anhductn2001 . Can we please see an example run from the fork?

Also, it might be me not understanding how this works exactly but how can we distinguish the logs between the various node containers?

For example, with upgrade testing, there are 2 chains, and each chain has 4 nodes running. Each node has a docker container associated with it.

We need a way to see the logs of each node container individually. Each container's logs are saved in its own file.

I don't have specific suggestions on the steps because this is a task that has a research component in it. However, the end goal is to have something like I described above.

**Comment by anhductn2001** [comment]
Yess, we can see an example here: https://github.com/anhductn2001/osmosis/actions/runs/4022809583
Logs have been saved in Artifacts: logs.tgz 
image
See the logs of each node container individually when we unzip it:
image

**Comment by p0mvn** [comment]
This is amazing. Great work.

Can you please add documentation here under "Debugging" section and then we can merge this:
https://github.com/osmosis-labs/osmosis/tree/main/tests/e2e#debugging

**Comment by anhductn2001** [comment]
I think it's done @p0mvn. We can merge it

File: tests/e2e/README.md [hunk 1]
```diff
@@ -254,4 +254,8 @@ This section contains information about debugging osmosis's `e2e` tests.
         docker logs < container name/id >
     ```
 
-    Example: `docker logs osmo-test-a-node-prune-nothing-snapshot` will print logs emitted by container `osmo-test-a-node-prune-nothing-snapshot` to your standard output.
\ No newline at end of file
+    Example: `docker logs osmo-test-a-node-prune-nothing-snapshot` will print logs emitted by container `osmo-test-a-node-prune-nothing-snapshot` to your standard output.
+
+3. Viewing docker container logs when run osmosis's `e2e` tests in CI
+
+    When a failure occurs in CI, we have persists these log files of all container running as artifacts ####logs.tgz.
```
Comment (p0mvn) [inline tests/e2e/README.md]:
```suggestion
    When a failure occurs in CI, we have persists these log files of all container running as artifacts file logs.tgz.
```

**Review by p0mvn — APPROVED** [review]

File: tests/e2e/README.md (hunk 1 above)
Comment (anhductn2001) [reply tests/e2e/README.md]: It's outdated @p0mvn. I have a new commit after that

**Pull request merged**